File: simpleMobileVMApp/simpleMobileVMAndroidLib/src/androidTest/java/com/slickpath/mobile/android/simple/vm/machine/FileHelperForTest.java
package com.slickpath.mobile.android.simple.vm.machine;

import com.slickpath.mobile.android.simple.vm.FileHelper;

public class FileHelperForTest implements FileHelper {

    private final String instructions;

    @SuppressWarnings("SameParameterValue")
    public FileHelperForTest(String instructions) {
        this.instructions = instructions;
    }

    @Override
    public String getInstructionsString() {
        return instructions;
    }
}

File: simpleMobileVMApp/simpleMobileVMAndroidLib/src/androidTest/java/com/slickpath/mobile/android/simple/vm/machine/FileHelperForTest.kt
package com.slickpath.mobile.android.simple.vm.machine

import com.slickpath.mobile.android.simple.vm.ParserHelper

class FileHelperForTest(private val instructionSetString: String) : ParserHelper {

    override fun getInstructionsString(): String {
        return instructionSetString
    }
}